File: Day068_2025/Q068_GFG.py
class Solution:
    def countPS(self, s: str) -> int:
        n = len(s)
        count = 0

        for center in range(n):
            # Odd length palindromes
            count += self.countPalindromes(s, center, center)
            # Even length palindromes
            count += self.countPalindromes(s, center, center + 1)

        return count

    def countPalindromes(self, s, left, right):
        count = 0
        while left >= 0 and right < len(s) and s[left] == s[right]:
            count += 1
            left -= 1
            right += 1
        return count

File: Day068_2025/Q068_GFG.java
class Q068_GFG {
    public int countPS(String s) {
        int n = s.length();
        int count = 0;

        for (int center = 0; center < n; center++) {
            // Odd length palindromes (centered at `center`)
            count += countPalindromes(s, center, center);
            // Even length palindromes (centered between `center` and `center + 1`)
            count += countPalindromes(s, center, center + 1);
        }

        return count;
    }

    private int countPalindromes(String s, int left, int right) {
        int count = 0;
        while (left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)) {
            count++;
            left--;
            right++;
        }
        return count;
    }
}
